shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/dependencyInjection/AppModule.kt

```diff
@@ -38,10 +39,15 @@ fun appModule(appVariant: AppVariant) = module {
 
 fun cacheModule() = module {
     factory { params ->
-        if (params.isNotEmpty()) {
-            ResponseCache(maxAge = params.get())
+        if (params.size() == 2) {
+            ResponseCache(cacheKey = params.get(), maxAge = params.get())
+        } else if (params.size() == 1) {
+            ResponseCache(cacheKey = params.get())
         } else {
-            ResponseCache()
+            throw InstanceCreationException(
+                "Wrong number of arguments provided to cacheModule",
+                parent = Exception()
+            )
```

Comment:
I wish there was a cleaner way to deal with named params in koin, like I said above, maybe it's not worth having this injected at all.

Reply:
It's probably not worth injecting it - there's no `IResponseCache` that we could use to implement a fake cache, and even if there was, the only place we'd use it is in `GlobalRepositoryTest` so we could easily just make it a parameter to `GlobalRepository` that defaults to a new real cache. If we wanted to, we could define and inject an explicit `ResponseCacheFactory`, but I'm not sure that'd solve a problem we actually have.

Reply:
Well, in the future we'll use it in at least one more repo, possibly more. But yeah I don't hate the idea of having it be a default param. We could also potentially make a separate koin module for each repository and key (though if we have a repo that programmatically generates keys with a param, it'll need to pass that in as a koin param still), or if we make it generic, each possible data type.